python/ml4ir/features/feature_config.py

```diff
@@ -252,24 +256,39 @@ def get_group_metrics_keys(self, key: str = None):
         """
         return self._get_list_of_keys_or_dicts(self.group_metrics_keys, key=key)
 
-    def define_inputs(self, max_num_records: int) -> Dict[str, Input]:
+    def get_dtype(self, feature_info: dict):
```

Comment:
@jakemannix Defining `get_dtype` and `get_default_value` under FeatureConfig. I've explicitly assigned dtypes to each tensor from this PR. This is necessary since I have to do some dynamic slicing on tensors later on. We should have always been doing this anyways.